build/media-suite_compile.sh

```diff
@@ -1497,53 +1497,74 @@ if enabled libsrt && do_vcs "https://github.com/Haivision/srt.git"; then
-        do_vcs "https://github.com/vapoursynth/vapoursynth.git"
-        do_uninstall {vapoursynth,vsscript}.lib "${_check[@]}"
-        if git show-ref -q "R${vsversion}"; then
-            git reset -q --hard "R${vsversion}"
-        else
-            git reset -q --hard origin/master
-        fi
+if  { ! mpv_disabled vapoursynth || enabled vapoursynth; }; then
+
+    _ver=3.7.3
+    _file=python37
+    [[ $bits = 64bit ]] && arch=amd64 || arch=win32
+    _check=("lib$_file.a")
+    if files_exist "${_check[@]}"; then
+        do_print_status "python $_ver" "$green" "Up-to-date"
+    elif do_wget "https://www.python.org/ftp/python/$_ver/python-$_ver-embed-$arch.zip"; then
+        create_build_dir
+        gendef "../$_file.dll" >/dev/null 2>&1
+        dlltool -y "lib$_file.a" -d "$_file.def"
+        [[ -f "lib$_file.a" ]] && do_install "lib$_file.a"
+        do_checkIfExist
+    fi
 
-        do_install include/*.h include/vapoursynth/
+    _vsver=46
+    _check=(lib{vapoursynth,vsscript}.a vapoursynth{,-script}.pc vapoursynth/{VS{Helper,Script},VapourSynth}.h)
+    if pc_exists "vapoursynth = $_vsver" && files_exist "${_check[@]}"; then
+        do_print_status "vapoursynth R$_vsver" "$green" "Up-to-date"
+    elif do_wget "https://github.com/vapoursynth/vapoursynth/releases/download/R$_vsver/VapourSynth${bits%bit}-Portable-R$_vsver.7z"; then
+        do_uninstall {vapoursynth,vsscript}.lib "${_check[@]}"
+        do_install sdk/include/*.h include/vapoursynth/
 
         create_build_dir
-        for _file in vapoursynth vsscript; do
-            gendef - "$vsprefix/${_file}.dll" 2>/dev/null |
-                sed -r -e 's|^_||' -e 's|@[1-9]+$||' > "${_file}.def"
-            dlltool -l "lib${_file}.a" -d "${_file}.def" \
-                $([[ $bits = 32bit ]] && echo "-U") 2>/dev/null
-            [[ -f lib${_file}.a ]] && do_install "lib${_file}.a"
-        done
-
-        for _file in vapoursynth{,-script}.pc; do
-            sed -e "s;@prefix@;$LOCALDESTDIR;" \
-                -e 's;@exec_prefix@;${prefix};' \
-                -e 's;@libdir@;${prefix}/lib;' \
-                -e 's;@includedir@;${prefix}/include;' \
-                -e "s;@VERSION@;$vsversion;" \
-                -e '/Libs.private/ d' \
-                -e '/Requires.private/ d' \
-                -e 's;lvapoursynth-script;lvsscript;' \
-                "../pc/$_file.in" > "$_file"
-                do_install "$_file"
+        for _files in vapoursynth vsscript; do
+            gendef "../$_files.dll" >/dev/null 2>&1
+            dlltool -y "lib$_files.a" -d "$_files.def"
+            [[ -f lib$_files.a ]] && do_install "lib$_files.a"
         done
 
+        printf "%s\n" "prefix=$LOCALDESTDIR" > vapoursynth.pc
+        printf "%s\n" "prefix=$LOCALDESTDIR" > vapoursynth-script.pc
+
+        sed -i \
+            -e 'a exec_prefix=${prefix}' \
+            -e 'a libdir=${exec_prefix}/lib' \
+            -e 'a includedir=${prefix}/include/vapoursynth' \
```

Comment:
These three lines are the same for both files. It's easier to make sure they are the same if you use a loop for both files instead of copypasting them and maintaining both separately.